serverless/src/git.ts

```diff
@@ -0,0 +1,41 @@
+export const filterSensitiveInfoFromRepository = (repositoryUrl: string | undefined) => {
+  try {
+    if (!repositoryUrl) {
+      return repositoryUrl;
+    }
+    if (repositoryUrl.startsWith("git@")) {
+      return repositoryUrl;
+    }
+    const { protocol, hostname, pathname } = new URL(repositoryUrl);
+    if (!protocol || !hostname) {
+      return repositoryUrl;
+    }
+
+    return `${protocol}//${hostname}${pathname}`;
```

Comment:
Do we have a test for this line?

Reply:
Good question, I will add some tests for filterSensitiveInfoFromRepository